File: examples/push/push.rb
require 'rubygems'
require 'pusher'

Pusher.app_id = 2192
Pusher.app_id = '2192'
Pusher.key = 'c4d8da3b6c36091f0ac4'
Pusher.secret = '50d3a48cd3e7133e7f55'

Pusher['test_channel'].trigger('my_event', {:user => 'pablo', :state => "happy", :id => 1})
sleep 1
Pusher['test_channel'].trigger('my_event', {:user => 'luis', :state => "sad", :id => 2})
sleep 1
Pusher['test_channel'].trigger('my_event', {:user => 'ana', :state => "tall", :id => 3})
sleep 1
Pusher['test_channel'].trigger('my_event', {:user => 'ana', :state => "cool", :id => 3})
sleep 1
Pusher['test_channel'].trigger('my_event', {:id => 1, :_action => "unset"})

File: examples/push/push.js
// Enable pusher logging - don't include this in production
Pusher.log = function() {
  if (window.console) window.console.log.apply(window.console, arguments);
};

// Flash fallback logging - don't include this in production
WEB_SOCKET_DEBUG = true;

var pusher = new Pusher('c4d8da3b6c36091f0ac4');
pusher.subscribe('test_channel');
pusher.bind('my_event', function(data) {
  string = JSON.stringify(data)
  console.log(string)
  users.updateFromJSON(string)

});

document.on("dom:loaded", function(){
  users = new Collection('users')
  users.template = "<li>User {{user}} is {{state}}</li>"
  users.set(1, {user: "pablo", state: "online"})
  users.set(2, {user: "denis", state: "offline"})
})
